tests/Integration/Behaviour/Features/Context/Domain/AbstractDomainFeatureContext.php

```diff
@@ -0,0 +1,98 @@
+<?php
+/**
+ * 2007-2019 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace Tests\Integration\Behaviour\Features\Context\Domain;
+
+use Behat\Behat\Context\Context;
+use Behat\Testwork\Hook\Scope\BeforeSuiteScope;
+use ObjectModel;
+use PrestaShop\PrestaShop\Core\CommandBus\CommandBusInterface;
+use RuntimeException;
+use Tests\Integration\Behaviour\Features\Context\CommonFeatureContext;
+use Tests\Integration\Behaviour\Features\Context\SharedStorage;
+
+abstract class AbstractDomainFeatureContext implements Context
+{
+    /**
+     * @var \Exception|null
+     */
+    protected $lastException;
+
+    /**
+     * @BeforeSuite
+     */
+    public static function prepare(BeforeSuiteScope $scope)
+    {
+        // Disable legacy object model cache to prevent conflicts between scenarios.
+        ObjectModel::disableCache();
```

Comment:
You fear nothing!

Reply:
It actually make sense to disable cache capabilities for such tests.
The funny thing is, I'm sure that it will not disable 100% of the cache 😛 I bet there is some cannot-be-disabled cache behaviors

Reply:
I actually copied this code from the develop branch without even reading it 😅
But I can guarantee you @matks that it is indeed not enough.. for example I had to call `Currency::getIdByIsoCode` with a `noCache` parameter to avoid unwanted behaviour Although in that case the `Cache` class was used

That said although disabling the cache like this is handy, it may give us false positive tests, because everything will work fine in the test but when cache is enabled in real context then unexpected behaviour will occur I don't really know what the ObjectModel cache is supposed to do so I can't imagine the consequences, but maybe it would be safer to disable it case by case when we really need it than disabling it altogether?

Reply:
Disabling the cache means it's not the same behavior as the front :thinking: 
Hope everything's going well!